src/methods/properties.js

```diff
@@ -995,6 +995,11 @@ export class PropertiesImplementation {
 
     // 6. Else if IsPropertyReference(V) is true, then
     if (Environment.IsPropertyReference(realm, V)) {
+      if (base instanceof AbstractValue) {
+        // Ensure that abstract values are coerced to objects. This might yield
+        // an operation that might throw.
+        base = To.ToObject(realm, base);
+      }
```

Comment:
This ensures that if we throw a fatal if we're not in pure mode already. It is the equivalent for what _dereference does. https://github.com/facebook/prepack/blob/88fbdcc47b15a38ac4369b7eeb893e2d62eedba2/src/methods/environment.js#L137-L141